hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -977,6 +979,11 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile machine approver: %w", err)
 	}
 
+	// Reconcile the network policies
+	if err = r.reconcileNetworkPolicies(ctx, createOrUpdate, hcluster); err != nil {
```

Comment:
Don't we also need to watch them?

Reply:
And looking at it, there is a lot of stuff we should be watching but don't. I'll try to come up with a unittest for that